modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -1787,6 +1787,17 @@ public CodegenModel fromModel(String name, Schema schema) {
             Map<String, Schema> allProperties = new LinkedHashMap<String, Schema>();
             List<String> allRequired = new ArrayList<String>();
 
+            // if schema has properties outside of allOf/oneOf/anyOf also add them to m
+            if (schema.getProperties() != null) {
+                addVars(m, unaliasPropertySchema(schema.getProperties()), schema.getRequired(), null, null);
```

Comment:
@wing328 See my [other comment](https://github.com/OpenAPITools/openapi-generator/pull/4482/files#r353522507) in this PR. Constraints around structures containing `oneOf` aren't very clearly defined in OAS 3.x, since the extension to include `discriminator` makes it look like any Schema properties are valid.

Do you think we should log a warning here, or wrap it the a ! strict spec check, or both? I've commented elsewhere as well that we might want to add to the validate and/or recommends about this.